shared/js/media/video_player.js

```diff
@@ -396,111 +399,83 @@ function VideoPlayer(container) {
-    var width, height; // The size the video will appear, after rotation
-    switch (rotation) {
-    case 0:
-    case 180:
-      width = givenWidth;
-      height = givenHeight;
-      break;
-    case 90:
-    case 270:
-      width = givenHeight;
-      height = givenWidth;
-    }
-
-    var xscale = containerWidth / width;
-    var yscale = containerHeight / height;
-    var scale = Math.min(xscale, yscale);
-
-    // Scale large videos down, and scale small videos up.
-    // This might reduce image quality for small videos.
-    width *= scale;
-    height *= scale;
-
-    var left = ((containerWidth - width) / 2);
-    var top = ((containerHeight - height) / 2);
-
-    var transform;
-    switch (rotation) {
-    case 0:
-      transform = 'translate(' + left + 'px,' + top + 'px)';
-      break;
-    case 90:
-      transform =
-        'translate(' + (left + width) + 'px,' + top + 'px) ' +
-        'rotate(90deg)';
-      break;
-    case 180:
-      transform =
-        'translate(' + (left + width) + 'px,' + (top + height) + 'px) ' +
-        'rotate(180deg)';
-      break;
-    case 270:
-      transform =
-        'translate(' + left + 'px,' + (top + height) + 'px) ' +
-        'rotate(270deg)';
-      break;
-    }
-
-    transform += ' scale(' + scale + ')';
-    return transform;
-  }
+  // Make the poster img element and video element fit the container
+  function setPlayerSize() {
+    // If we don't have an aspectRatio, then the load() method has not
+    // been called yet (or the reset() method has already been called).
+    // There is nothing we can do in that case (but we might still be called
+    // because of a resize event handler)
+    if (!aspectRatio) {
+      return;
+    }
 
-  // Make the video fit the container
-  function setPlayerSize(playerOnly) {
+    // Given the size of the container, and the aspect ratio of the video
+    // (and of its poster image) we need to figure out the largest possible
+    // playback size for the video. (Note that he actual size of the video
+    // and the poster image is does not matter: the img and video elements
+    // will scale them to the playback size for us.)
     var containerWidth = container.clientWidth;
     var containerHeight = container.clientHeight;
 
-    // Don't do anything if we don't know our size.
-    // This could happen if we get a resize event before our metadata loads
-    if (!videowidth || !videoheight || !posterwidth || !posterheight) {
-      return;
+    // The video size doesn't matter, so let's pick an arbitrary width
+    // to make these computations easier.
+    var dummyVideoWidth = 1080;
+    var dummyVideoHeight = dummyVideoWidth / aspectRatio;
+    var scale = Math.min(containerWidth / dummyVideoWidth,
+                         containerHeight / dummyVideoHeight);
+
+    // These are the maximum playback dimensions
+    playbackWidth = Math.round(dummyVideoWidth * scale);
+    playbackHeight = Math.round(dummyVideoHeight * scale);
```

Comment:
Love the size independent scaling of the poster/video elements; never struck my mind. I think the logic above can be simplified to avoid division and reduce the number (and at least to be me, easier to understand what's going on):

    // If the container aspect ratio is larger than the poster/video
    // aspect ratio, then the playback size is constrained by the height
    // in order to display the entire content without clipping. If it is
    // smaller, then it is constrained by the width.
    if (containerWidth > aspectRatio * containerHeight) {
      playbackWidth = Math.round(containerHeight * aspectRatio);
      playbackHeight = containerHeight;
    } else {
      playbackWidth = containerWidth;
      playbackHeight = Math.round(containerWidth / aspectRatio);
    }